include/analog_digital_converter.h

```diff
@@ -0,0 +1,85 @@
+#ifndef ADC_H
+#define ADC_H
+
+#include "system.h"
+
+/**
+ * @brief A driver for the analog-digital converters available on the microcontroller.
+ * @details Besides initialization, it is most convenient to work with the ADC with the
+ * AnalogPin class, which handles adding channels and reading results.
+ */
+class AnalogDigitalConverter
+{
+public:
+  /**
+   * @brief Initializes the ADC according to the definition struct provided
+   * @param A hardware struct defining the resources for the ADC
+   */
+  void init(const ADCHardwareStruct *adc_def);
+  /**
+   * @brief Adds a channel to the list that is scanned.
+   * @details This returns the rank assigned to the channel, which is needed to read it later
+   * This method does not check for errors. Do not add more than CHANNEL_COUNT channels,
+   * by calling add_channel more than CHANNEL_COUNT times
+   * @param channel The index of the channel to be read
+   * @return The rank of the added channel
+   */
+  uint8_t add_channel(uint8_t channel);
+  // Returns a value between 0 and RAW_READING_MAX. 0 represents 0V, and RAW_READING_MAX
+  // represents REFERENCE_VOLTAGE
+  // The parameter, rank, is the return value from add_channel when the channel was added
+  /**
+   * @brief Reads a single channel.
+   * @details The reading is between 0 and RAW_READING_MAX. 0 represents 0V, and
+   * RAW_READING_MAX represents REFERENCE_VOLTAGE (typically 3.3V).
+   * Takes the rank of the channel as a parameter, which is returned from add_channel
+   * @param rank The rank of the channel to be read.
+   * @return
+   */
+  uint16_t read(uint8_t rank) const;
```

Comment:
is `rank` a term from the datasheet?  I feel like `index` or `channel_id` might be easier to understand

Reply:
I renamed these to index.